src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -143,11 +150,33 @@ public Set<AccessProfile> generateAccessProfilesByCaseReference(String caseRefer
         FilteredRoleAssignments filteredRoleAssignments =
             roleAssignmentsFilteringService.filter(roleAssignments, caseDetails.get());
 
-
         CaseTypeDefinition caseTypeDefinition = caseDefinitionRepository.getCaseType(caseDetails.get().getCaseTypeId());
 
-        return Sets.newHashSet(filteredAccessProfiles(filteredRoleAssignments.getFilteredMatchingRoleAssignments(),
-            caseTypeDefinition, false));
+        List<AccessProfile> filteredAccessProfiles = filteredAccessProfiles(
+            filteredRoleAssignments.getFilteredMatchingRoleAssignments(), caseTypeDefinition, false);
+
+        final String caseAccessCategory = getCaseAccessCategoryFromCaseData(caseDetails.get());
+
+        return filterAccessProfilesByCaseAccessCategory(filteredAccessProfiles, caseAccessCategory);
```

Comment:
Minor: These three lines are almost identical to the three in the previous method - worth extracting?

Reply:
I thought about this, then decided to wait for the third method which is in the  PR now.